Packs/CortexAttackSurfaceManagement/Scripts/RemediationPathRuleEvaluation/RemediationPathRuleEvaluation.py

```diff
@@ -0,0 +1,225 @@
+from typing import Any, Dict, List
+
+# should match IncidentSeverity in CommonServerPython
+SEVERITY_MAP = {
+    "unknown": 0,
+    "info": 0.5,
+    "low": 1,
+    "medium": 2,
+    "high": 3,
+    "critical": 4,
+}
+
+
+def match_remediation_rule(alert_context: Dict[str, Any], rules: List) -> List:
+    """
+    For the given alert (with the field in alert_context), finds the most recently created
+    remediation path rule where the alert meets the rule's criteria
+
+    Args:
+        alert_context: dictionary of relevant alert fields to check against
+        rules: list of remediation path rules to evaluate
+
+    Returns: the matching rule (list of 1)
+
+    """
+    matched_rules = []
+    # loop through remediation rules
+    if isinstance(rules, dict):
+        rules = [rules]
+    for rule in rules:
+        match = True
+        conditions = rule["criteria"]
+        if len(conditions) == 0:
+            demisto.info(
+                "Invalid Remediation Path Rule - no criteria condition to evaluate."
+            )
+            match = False
+        else:
+            # check criteria conjunction
+            if rule.get("criteria_conjunction") == "AND":
+                for cond in conditions:
+                    rule_matched = evaluate_criteria(cond, alert_context)
+                    if not rule_matched:
+                        match = False
+            else:
+                demisto.info(
+                    f"Criteria conjunction {rule.get('criteria_conjunction')} not supported at this time."
+                )
+                match = False
+
+        if match:
+            matched_rules.append(rule)
+
+    # sort matched rules by created_at timestamp in descending order
+    # return the most recent one
+    matched_rules_sorted = sorted(
+        matched_rules, key=lambda x: x["created_at"], reverse=True
+    )
+    if len(matched_rules_sorted) > 0:
+        return [matched_rules_sorted[0]]
+    else:
+        return []
+
+
+def evaluate_criteria(cond: Dict, alert_context: Dict) -> bool:
+    """
+    Evaluate whether a criteria condition is met based on the given alert's context fields
+
+    Args:
+        cond: Condition to evaluation (dictionary with keys: field, operator, value)
+        alert_context: dictionary of relevant alert fields to check against
+
+    Returns: True if the condition is met or False if not
+
+    """
+    field = cond["field"]
+    operator = cond["operator"]
+    value = cond["value"]
+
+    alert_context_value = alert_context.get(field)
+
+    # force lowercase for field and operator and value
+    if isinstance(field, str) and isinstance(operator, str) and isinstance(value, str):
+        field = field.lower()
+        operator = operator.lower()
+        value = value.lower()
+    else:
+        demisto.info("Invalid condition: field, operator or value must be strings")
+        return False
+
+    if operator == "eq":
+        # severity
+        if field == "severity":
```

Comment:
It might help us to reduce complexity here in case it gets bigger in the future. Is it possible to make this nested `if field == "severity"` statement not nested by moving down. That way we could use something like:

```suggestion
    if operator != "eq":
        demisto.info(f"Condition Operator {operator} is not supported at this time.")
        return False
```
and then it can exit early.

Reply:
We can change this later if needed but for now all operators are "eq" so we shouldn't see it match on anything else for now.